pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -493,7 +492,7 @@ def _do_save(self, trainer: 'pl.Trainer', filepath: str) -> None:
         trainer.dev_debugger.track_checkpointing_history(filepath)
 
         # make paths
-        if trainer.is_global_zero or tpu_training_and_local_rank_zero(trainer):
+        if trainer.should_save_checkpoint:
```

Comment:
we should also drop the `@rank_zero_only` decorator inside this callback and use this trainer property for everything to be consistent.

cc @shuyingsunshine21

Reply:
@ananthsub Good catch! Updated